src/adapter/src/explain_new/hir/text.rs

```diff
@@ -98,13 +97,23 @@ impl<'a> DisplayText<PlanRenderingContext<'_, HirRelationExpr>>
                     }
                 },
                 Project { outputs, input } => {
-                    let outputs = Indices(outputs);
-                    writeln!(f, "{}Project {}", ctx.indent, outputs)?;
+                    if outputs.is_empty() {
+                        // projecting an empty tuple is allowed
+                        writeln!(f, "{}Project", ctx.indent)?;
+                    } else {
+                        let outputs = Indices(outputs);
+                        writeln!(f, "{}Project {}", ctx.indent, outputs)?;
+                    }
                     ctx.indented(|ctx| Displayable::from(input.as_ref()).fmt_text(f, ctx))?;
                 }
                 Map { scalars, input } => {
-                    let scalars = separated_text(", ", scalars.iter().map(Displayable::from));
-                    writeln!(f, "{}Map {}", ctx.indent, scalars)?;
+                    if scalars.is_empty() {
+                        // mapping without producing a columns is (probably?) allowed
```

Comment:
> I don't think this allowed because our Map generator function suppresses empty maps and automatically fuses consecutive maps.

This can only be considered hard evidence if we were not able to construct `Map { ... }` directly, which at the moment is not the case, so I would rather keep this case.